internal/service/searchindex/resource_search_index.go

```diff
@@ -187,79 +187,78 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	projectID := ids["project_id"]
 	clusterName := ids["cluster_name"]
 	indexID := ids["index_id"]
+	indexName := d.Get("name").(string)
 
-	searchIndex, _, err := connV2.AtlasSearchApi.GetAtlasSearchIndex(ctx, projectID, clusterName, indexID).Execute()
+	if d.HasChange("name") || d.HasChange("type") || d.HasChange("database") || d.HasChange("collection_name") {
+		return diag.Errorf("error updating search index (%s): attributes name, type, database and collection_name can't be updated", indexName)
+	}
+
+	searchRead, _, err := connV2.AtlasSearchApi.GetAtlasSearchIndex(ctx, projectID, clusterName, indexID).Execute()
 	if err != nil {
 		return diag.Errorf("error getting search index information: %s", err)
 	}
-
-	if d.HasChange("type") {
-		searchIndex.Type = conversion.StringPtr(d.Get("type").(string))
+	searchIndex := &admin.SearchIndexUpdateRequest{
+		Definition: admin.SearchIndexUpdateRequestDefinition{
+			Analyzer:       searchRead.LatestDefinition.Analyzer,
+			Analyzers:      searchRead.LatestDefinition.Analyzers,
+			Mappings:       searchRead.LatestDefinition.Mappings,
+			SearchAnalyzer: searchRead.LatestDefinition.SearchAnalyzer,
+			StoredSource:   searchRead.LatestDefinition.StoredSource,
+			Synonyms:       searchRead.LatestDefinition.Synonyms,
+			Fields:         searchRead.LatestDefinition.Fields,
+		},
```

Comment:
if we dont include these properties in the PATCH do we face unexpected changes?

Reply:
we were doing this before.  
I tried not to do it but i got errors because the API expects the properties even if they didn't change